src/raiden_libs/service_registry.py

```diff
@@ -273,32 +295,44 @@ def maybe_promt(query: str) -> None:
 
         # mint tokens if necessary, but only on testnets
         if account_balance < latest_deposit and web3.eth.chainId != 1:
-            checked_transact(
+            receipt = checked_transact(
                 web3=web3,
                 sender_address=service_address,
                 function_call=deposit_token_contract.functions.mint(latest_deposit),
                 task_name="Minting new Test RDN tokens",
+                wait_until_block=wait_until_block,
             )
 
+            wait_until_block = receipt["blockNumber"]
```

Comment:
Shouldn't this be 
```suggestion
            wait_until_block = max(receipt["blockNumber"], wait_until_block)
```
?

Reply:
Isn't it per definition always the case that  `receipt["blockNumber"] >= wait_until_block`?

Reply:
Yes it is, because `checked_transact` blocks until the transaction is mined. Ignore this.